src/wp-includes/class-wp-theme.php

```diff
@@ -363,10 +376,11 @@ public function __construct( $theme_dir, $theme_root, $_child = null ) {
 				$this->cache_add(
```

Comment:
Same here: This conditional above already checks that the templates are missing, so by definition we know it's not a block theme.
```suggestion
				$this->block_theme = false;
				$this->cache_add(
```